src/global/BaseTest.java

```diff
@@ -39,12 +39,23 @@ public String getItemAtIndex(int index) {
         try {
             if (matcher.find()) return matcher.group(index);
             else fail("Your code's output did not follow the correct structure/syntax");
-        } catch (ArrayIndexOutOfBoundsException e) {
-            fail("The specified group doesn't exist");
+        } catch (IndexOutOfBoundsException e) {
+            fail("The specified group (" + (index) + ") doesn't exist");
         }
         return "";  // TODO: logically how does this behave?
     }
 
+    public String getItemByName(String name) {
```

Comment:
Each Clause already has a built-in function to cast the string from the group thing into the correct type it needs to be. I think it'd be super beneficial to carry that through so this method returns the correct type.

Idk how tough this is though because Java is baby

Reply:
opened #17